### ISSUE TITLE
Disabling exclusions in a lint() call

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1515]
Is it possible to disable linting in a single `lint()` call via arguments to `lint()`? I can only figure out how to do so via `options()`, and I don't see anything on [`?exclude`](https://lintr.r-lib.org/reference/exclude.html) to help.

**Comment by MichaelChirico** [comment]
For context, I am trying to run one specific linter on generic file input; the issue is that this can generate `warning()`s when the file uses `# nolint: some_linter_other_than_the_one_im_running`, so I'd like to pass some arguments to `lint()` to say "ignore exclusions".

**Comment by AshesITR** [comment]
I don't think this is possible atm.

A pragmatic solution would be to temporarily change the nolint regexes, or to use suppressWarnings() / withCallingHandlers() to (selectively) suppress the unwanted warnings.

**Comment by MichaelChirico** [comment]
Seems like the best approach is to expose something through lint(), all told.

for now, I'm using

```r
with_options(
  list(lintr.exclude='\1', lintr.exclude_start='\1'),
  lint(...)
)
```

but that's pretty hacky/ugly/obscure.

lint(..., exclusions=NULL) seems like a natural API for this. but need to think through how it interacts w/ everything else.
